controllers/users_test.go

```diff
@@ -130,3 +256,75 @@ func TestCreateUser(t *testing.T) {
 		})
 	}
 }
+
+func TestDeleteUser(t *testing.T) {
+	// Create test data
+	dataSet, err := createUserTestData(DeleteUserTest)
+	if err != nil {
+		t.Errorf("Failed to create test data %s", err)
+		return
+	}
+
+	// Run tests
+	for _, testCaseString := range dataSet.OrderedList {
+		testCaseString := testCaseString
+		t.Run(testCaseString, func(t *testing.T) {
+			testCase := dataSet.TestDataSet[testCaseString]
+			var expectedError error
+			if testCase.Expected.(map[string]interface{})["error"] != nil {
+				expectedError = testCase.Expected.(map[string]interface{})["error"].(error)
+			}
+
+			expectedUserDeleted := testCase.Expected.(map[string]interface{})["user_deleted"].(bool)
+			expectedProductDeleted := testCase.Expected.(map[string]interface{})["product_deleted"].(bool)
+			expectedUsersProducts := testCase.Expected.(map[string]interface{})["users_products_updated"].(bool)
+
+			userID := testCase.Data.(map[string]interface{})["user_id"].(uuid.UUID)
+			var nominatedOwners map[uuid.UUID]uuid.UUID
+			if testCase.Data.(map[string]interface{})["nominees"] != nil {
+				nominatedOwners = testCase.Data.(map[string]interface{})["nominees"].(map[uuid.UUID]uuid.UUID)
+			}
+
+			dbMockUser := testCase.Data.(map[string]interface{})["db_mock_user"].(*models.User)
+			dbMockProduct := testCase.Data.(map[string]interface{})["db_mock_product"].(*models.Product)
+			dbMockUsersProducts := testCase.Data.(map[string]interface{})["db_mock_users_products"].(models.UserProducts)
+
+			var dbMockError error
+			if testCase.Data.(map[string]interface{})["db_mock_error"] != nil {
+				dbMockError = testCase.Data.(map[string]interface{})["db_mock_error"].(error)
+			}
+
+			mysqldb.Functions = &DBFunctionInterfaceMock{
+				user:                 dbMockUser,
+				product:              dbMockProduct,
+				userProducts:         &dbMockUsersProducts,
+				err:                  dbMockError,
+				userDeleted:          test.NewBool(false),
+				productDeleted:       test.NewBool(false),
+				usersProductsUpdated: test.NewBool(false),
+				privileges:           testCase.Data.(map[string]interface{})["db_mock_privileges"].(models.Privileges),
+			}
+
+			err := DeleteUser(&userID, nominatedOwners)
+			if !test.ErrEqual(err, expectedError) {
+				t.Errorf(test.TestResultString, testCaseString, err, expectedError)
+				return
+			}
+
+			if !cmp.Equal(*mysqldb.Functions.(*DBFunctionInterfaceMock).userDeleted, expectedUserDeleted) {
```

Comment:
Would be better if could solve without this asterisk:
```suggestion
			if !cmp.Equal(mysqldb.Functions.(*DBFunctionInterfaceMock).userDeleted, expectedUserDeleted) {
```

Reply:
#22 is created to fix all unnecessary pointer handling in mocks, since changing to pointer receivers.